text/0000-build-std.md

```diff
@@ -0,0 +1,2390 @@
+  - [cargo#10308] is a prototype with explicit standard library dependencies
+    - Idea posted on internals, see *Build-std and the standard library* below
+  - If enabled manually, [cargo#8733] expressed the need to enable build-std
+    only for certain targets, in particular when cross-compiling to no-std
+    targets
+- **[wg-cargo-std-aware#46]: How to handle special object files?**, [ehuss], Sep 2019
+  - How to handle pre-built object files needed to link on a target?
+    - e.g. `crt1.o`, `crti.o`, `crtn.o`, etc for musl or `dllcrt2.o`,
+      `crt2.o`, etc for Windows GNU
+  - Cargo is target agnostic and does not want to hardcode logic for particular
+    targets
+    - The target definitions define which files they expect to find and link
+      against
+  - [rust#68887] adds a `self-contained` directory which could help
+- **[wg-cargo-std-aware#47]: how to handle pre-built linkers?**, [ehuss], Sep
+  2019
+  - Some pre-built targets ship with a copy of rust-lld or gcc to assist with
+    linking
+  - `rust-lld` could be shipped as a rustup component or the user could be
+    forced to install these components
+    - Since this issue was filed, `rust-lld` is now always shipped
+  - rustc finds `rust-lld` via `--sysroot` and if the sysroot is not provided
+    then another mechanism will need to be used to find `rust-lld`
+- **[wg-cargo-std-aware#50]: Impact on build scripts that invoke rustc**, [jdm], Oct 2019
+  - Need to make sure that build scripts that invoke rustc are able to do this with the correct
+    standard library
+  - Closed as t-cargo do not want to encourage or support build probes
+- **[wg-cargo-std-aware#51]: Plan for removal of `rustc-dep-of-std`**, [ehuss], Nov 2019
+  - `rustc-dep-of-std` is a feature that packages used as dependencies of `std` have that,
+    when enabled, use to declare explicit dependencies of `core`/`compiler_builtins`/etc
+    - May not be necessary once these packages are always declaring dependencies on
+      `core`/etc
+    - Cargo would still need to work out what to do when seeing explicit dependencies
+  - These dependencies could always be built with build-std but that would involve
+    using build-std in bootstrap which may not be desirable
+  - Alternatively, have some mechanism for telling Cargo that the explicit dependencies
+    in this instance aren't from the sysroot but an rlib from bootstrap
+    - Could be automatic or variation on patch syntax
+- **[wg-cargo-std-aware#57]: Support building workspace with separate build-std
+  parameters**, [jschwe], Jun 2020
+  - User wants to have one package depend on `core` and `alloc` and another
+    package depend on `core`, `alloc` and `std` but this isn't possible with the
+    current experimental flag
+  - Closed as duplicate of [wg-cargo-std-aware#5]
+- **[wg-cargo-std-aware#68]: Support Profile Guided Optimisation (PGO)**, [errantmind], Mar 2021
+  - `profiler_builtins` isn't compatible with `no_core`
```

Comment:
This may have changed in the interim (eg, [`profiler_builtins` is `#![no_core]`](https://github.com/rust-lang/rust/blob/0d6ab209c525e276cbe7544cbd39a3c3619b6b18/library/profiler_builtins/src/lib.rs#L9)).